cipher/ecc.c

```diff
@@ -1613,10 +1650,35 @@ ecc_decrypt_raw (gcry_sexp_t *r_plain, gcry_sexp_t s_data, gcry_sexp_t keyparms)
       goto leave;
     }
 
-
   ec = _gcry_mpi_ec_p_internal_new (sk.E.model, sk.E.dialect, flags,
                                     sk.E.p, sk.E.a, sk.E.b);
 
+  /* For GOST extract the UKM value. */
+  if ((flags & PUBKEY_FLAG_GOST) || 0 == strncmp ("GOST", sk.E.name, 4))
```

Comment:
It's a bad idea to depend on exact curve name.

Reply:
* `strNcmp()` isn't an exact comparison but a prefix check.
* Do you mean it is necessary to check `sk.E.name` to be an OID?

Reply:
No. It is just a bad idea to depend on the curve names. It can be an OID. One can use it with any other curve. Etc.

Reply:
Dropped `strncmp()`. Only the `flags` now.